database/src/main/postgres/flows/V1.7.5__get_flow_checkpoints.sql

```diff
@@ -0,0 +1,127 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+CREATE OR REPLACE FUNCTION flows.get_flow_checkpoints(
+    IN i_partitioning_of_flow JSONB,
+    IN i_limit                INT DEFAULT 5,
+    IN i_checkpoint_name      TEXT DEFAULT NULL,
+    OUT status                INTEGER,
+    OUT status_text           TEXT,
+    OUT id_checkpoint         UUID,
+    OUT checkpoint_name       TEXT,
+    OUT measure_name          TEXT,
+    OUT measure_columns       TEXT[],
+    OUT measurement_value     JSONB,
+    OUT checkpoint_start_time TIMESTAMP WITH TIME ZONE,
+    OUT checkpoint_end_time   TIMESTAMP WITH TIME ZONE
+) RETURNS SETOF record AS
+$$
+-------------------------------------------------------------------------------
+--
+-- Function: flows.get_flow_checkpoints(3)
+--      Retrieves all checkpoints (measures and their measurement details) related to a primary flow
+--      associated with the input partitioning.
+--
+-- Note: a single row returned from this function doesn't contain all data related to a single checkpoint - it only
+--     represents one measure associated with a checkpoint. So even if only a single checkpoint would be retrieved,
+--     this function can potentially return multiple rows.
+--
+-- Note: checkpoints will be retrieved in ordered fashion, by checkpoint_time and id_checkpoint
+--
+-- Parameters:
+--      i_partitioning_of_flow  - partitioning to use for identifying the flow associate with checkpoints
+--                                that will be retrieved
+--      i_limit                 - (optional) maximum number of checkpoints retrieved
+--                                if zero or lower specified, all data will be returned, i.e. no limit will be applied
+--      i_checkpoint_name       - (optional) if specified, returns data related to particular checkpoint's name
+--
+--      Note: checkpoint name uniqueness is not enforced by the data model, so there can be multiple different
+--          checkpoints with the same name in the DB, i.e. multiple checkpoints can be retrieved even when
+--          specifying `i_checkpoint_name` parameter
+--
+-- Returns:
+--      status              - Status code
+--      status_text         - Status text
+--      id_checkpoint       - id of retrieved checkpoint
+--      checkpoint_name     - name of retrieved checkpoint
+--      measure_name        - measure name associated with a given checkpoint
+--      measure_columns     - measure columns associated with a given checkpoint
+--      measurement_value   - measurement details associated with a given checkpoint
+--      checkpoint_time     - time
+--
+-- Status codes:
+--      11                  - Flow created
```

Comment:
```suggestion
--      11                  - OK
```

Reply:
hmm these status texts in the code docs are not always 1:1 to real messages. So I just wanted to make it more human readable / understandable.

Reply:
Wait what?